orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/ecs/EcsServerGroupCreator.groovy

```diff
@@ -86,6 +129,51 @@ class EcsServerGroupCreator implements ServerGroupCreator, DeploymentDetailsAwar
     return [[(ServerGroupCreator.OPERATION): operation]]
   }
 
+  private Supplier<Iterable<Object>> fetchAndParseArtifact(Artifact artifact) {
+    return { ->
+      Response artifactText = oortService.fetchArtifact(artifact)
+      try {
+        return yamlParser.get().loadAll(artifactText.getBody().in());
```

Comment:
I'm conflicted about using a YAML parser on file that's expected as JSON. On one hand, YAML is ostensibly a superset of JSON, so it should work fine and would let customers use either format (though I'm not sure whether the current object mapper in clouddriver will work with a YAML artifact...). On the other hand, a dedicated JSON parser may be more performant and we'd have even more confidence of avoiding YAML/JSON spec incompatibilities (if there are any?).

Do we know of anywhere else in Orca where a JSON parser is being used instead of the YAML one? Also, can we test a YAML artifact with true/false `evaluateTaskDefinitionArtifact` values to confirm whether YAML is de facto supported now?

Reply:
Currently Clouddriver mapper doesn't support YAML parsing and ORCA does support it, which means if the evaluateTaskDefinitionArtifact flag is set as true, then YAML or JSON  artifact both will get evaluated and sent to clouddriver. If evaluateTaskDefinitionArtifact is set to false and YAML has been used as an artifact then it will fail in clouddriver. So best case would to add a support for YAML also in clouddriver and we can create a new issue to add this functionality. Does that make sense?